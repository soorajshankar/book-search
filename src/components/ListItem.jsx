import React from 'react'
import {trimString} from '../utils/UiUtil'

const ListItem = ({title = '', author = '', summary = ''}) => {
  const smr = trimString(summary, 200)
  return (
    <div className="c-list__item" data-testid="book_item">
      <h3 className="c-card__title">{title}</h3>
      <h4 className="c-card__author">{author}</h4>
      <p className="c-card__summary">{smr}</p>
    </div>
  )
}

export default ListItem
