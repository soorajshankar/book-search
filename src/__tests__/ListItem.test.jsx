import React from 'react'
import {render, fireEvent} from '@testing-library/react'
import ListItem from '../components/ListItem'

describe('renders ListItem', () => {
  it('should match Snapshot', () => {
    let count = 0
    const {getByText, container} = render(
      <ListItem author="author" title="title" summary="salkfnalns aslfkn" />,
    )
    expect(container).toMatchSnapshot()
  })
  it('should render with values', () => {
    const {getByText, container} = render(
      <ListItem author="author1" title="title1" summary="summary1" />,
    )
    expect(container.querySelector('.c-card__title').textContent).toEqual(
      'title1',
    )
    expect(container.querySelector('.c-card__author').textContent).toEqual(
      'author1',
    )
    expect(container.querySelector('.c-card__summary').textContent).toEqual(
      'summary1',
    )
  })
})
