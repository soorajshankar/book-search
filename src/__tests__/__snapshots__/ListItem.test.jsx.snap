// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders ListItem should match Snapshot 1`] = `
<div>
  <div
    class="c-list__item"
  >
    <h3
      class="c-card__title"
    >
      title
    </h3>
    <h4
      class="c-card__author"
    >
      author
    </h4>
    <p
      class="c-card__summary"
    >
      salkfnalns aslfkn
    </p>
  </div>
</div>
`;
